src/main/java/com/google/sps/servlets/DeleteTutorSessionServlet.java

```diff
@@ -44,12 +55,92 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         String studentID = Optional.ofNullable((String)request.getSession(false).getAttribute("userId")).orElse("-1");
         long id = Long.parseLong(request.getParameter("id"));
 
+        TutorSession tutorSession = datastore.getScheduledSession(id);
+
+        System.out.println(tutorSession.getStudentID());
+
+        Entity studentEntity = datastore.getStudentForUserId(tutorSession.getStudentID());
+        Entity tutorEntity = datastore.getTutorForUserId(tutorSession.getTutorID());
+        TimeRange timeslot = tutorSession.getTimeslot();
+
+        boolean testTutorEmail = sendCancellationEmailToTutor(studentEntity, tutorEntity, timeslot);
+        boolean testStudentEmail = sendCancellationEmailToStudent(studentEntity, tutorEntity, timeslot);
+
         // Remove tutor session
         datastore.deleteTutorSession(studentID, id);
 
         String json = new Gson().toJson(datastore.getScheduledSession(id));
         response.setContentType("application/json;");
-        response.getWriter().println(json);
+        response.getWriter().println(json + "testTutorEmail: " + testTutorEmail + " testStudentEmail: " + testStudentEmail);
         return;
     }
+
+    public boolean sendCancellationEmailToTutor(Entity studentEntity, Entity tutorEntity, TimeRange timeslot) {
+        Properties props = new Properties();
+        Session session = Session.getDefaultInstance(props, null);
```

Comment:
What is a `Session` here?

Reply:
It is basically the session accessed by the email host to create the message and send it out. Here's more info about that: https://stackoverflow.com/questions/22657918/what-is-the-use-of-javax-mail-session